common/src/test/java/org/apache/sedona/common/raster/RasterBandEditorsTest.java

```diff
@@ -69,6 +73,23 @@ public void testSetBandNoDataValueWithEmptyRasterMultipleBand() throws FactoryEx
         assertEquals(444, (double) RasterBandAccessors.getBandNoDataValue(grid, 2), 0.1d);
     }
 
+    @Test
+    public void testClip() throws IOException, FactoryException, TransformException, ParseException {
+        GridCoverage2D raster = rasterFromGeoTiff(resourceFolder + "raster_geotiff_color/FAA_UTM18N_NAD83.tif");
```

Comment:
Please add tests to try with `crop == true`. In addition, please also test if the pixels of a raster are successfully masked.